bn-apps/cordapp-updates-distribution/corda-updates-shell/src/main/kotlin/net/corda/businessnetworks/cordaupdates/shell/CordaUpdatesCLI.kt

```diff
@@ -0,0 +1,166 @@
+package net.corda.businessnetworks.cordaupdates.shell
+
+import com.github.mustachejava.DefaultMustacheFactory
+import net.corda.businessnetworks.cordaupdates.core.ArtifactMetadata
+import net.corda.businessnetworks.cordaupdates.core.CordaMavenResolverException
+import net.corda.businessnetworks.cordaupdates.core.CordappSyncer
+import net.corda.businessnetworks.cordaupdates.core.ResourceNotFoundException
+import net.corda.businessnetworks.cordaupdates.core.ResourceTransferException
+import net.corda.businessnetworks.cordaupdates.core.SyncerConfiguration
+import net.corda.cliutils.CordaCliWrapper
+import net.corda.cliutils.ExitCodes
+import net.corda.cliutils.start
+import org.eclipse.aether.artifact.DefaultArtifact
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import picocli.CommandLine
+import java.io.File
+import java.io.FileOutputStream
+import java.io.PrintWriter
+import java.nio.file.Files
+
+private enum class CLICommands {
+    INIT, SYNC, PRINT_VERSIONS
+}
+
+/**
+ * Class that handles CLI interactions. Supports the following modes via --mode flag:
+ * - INIT. Initializes an empty local repository and creates a template configuration file under USER.HOME/.corda-updates
+ * folder by default unless a custom location has been provided via --configPath flag
+ * - SYNC. Pulls down missing CorDapps from configured remote repositories. Downloads all versions of all CorDapps from all remote repositories,
+ * that are configured in the settings.conf file. If a custom CorDapp / version range has been specified via --cordapp flag, the tool will attempt to
+ * find a respective cordapp source in the settings.conf and will fail if one doesn't exist.
+ * - PRINT_VERSIONS. Prints available versions of a specified CorDapp to the console. Will fail if no CorDapp name has been provided.
+ */
+class CordaUpdatesCLI : CordaCliWrapper("corda-updates", "CLI for corda-updates utility.") {
```

Comment:
Is a separate tool really the way to go here? Why not have these as Corda shell commands?

Reply:
Yes, makes sense. I didn't know that corda shell is extensible. I excluded corda-shell to a separate PR.